api/cluster/virtual_service.go

```diff
@@ -0,0 +1,188 @@
+package cluster
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/url"
+	"strings"
+
+	istiov1beta1 "istio.io/api/networking/v1beta1"
+	v1beta1 "istio.io/client-go/pkg/apis/networking/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	"github.com/caraml-dev/merlin/models"
+	"github.com/caraml-dev/merlin/pkg/protocol"
+)
+
+const (
+	knativeIngressGateway          = "knative-serving/knative-ingress-gateway"
```

Comment:
Seems risky to hardcode these (what if knative / istio were installed in a different namespace?). But I realise we are doing the same thing in [Turing](https://github.com/caraml-dev/turing/blob/main/api/turing/cluster/servicebuilder/router.go#L82) and something similar is done in the kserve codebase too. Even if we don't deal with it now, I guess it would be better to add a TODO comment to make these configurable.